task/control-plane-agent/src/mgs_sidecar.rs

```diff
@@ -1042,3 +1133,36 @@ fn sp_error_from_ignition_error(err: IgnitionError) -> SpError {
     };
     SpError::Ignition(err)
 }
+
+fn get_ecdsa_challenge() -> Result<EcdsaSha2Nistp256Challenge, SpError> {
+    // Get a nonce from our RNG driver
+    let rng = drv_rng_api::Rng::from(RNG.get_task_id());
+    let mut buf = [0u8; 32];
+    rng.fill(&mut buf).map_err(|e| {
+        ringbuf_entry!(Trace::RngFillFailed(e));
+        SpError::Monorail(GwMonorailError::GetChallengeFailed)
+    })?;
+
+    // Get our hardware ID from Packrat
+    let packrat = task_packrat_api::Packrat::from(
+        crate::mgs_common::PACKRAT.get_task_id(),
+    );
+    let identity = packrat.get_identity().unwrap_or(VpdIdentity::default());
+    const HW_ID_LEN: usize = 32;
+    let mut hw_id = [0u8; HW_ID_LEN];
+    static_assertions::const_assert!(
+        HW_ID_LEN >= core::mem::size_of::<VpdIdentity>()
+    );
+    hw_id[..core::mem::size_of::<VpdIdentity>()]
+        .copy_from_slice(identity.as_bytes());
+
+    let now = sys_get_timer().now;
+    Ok(EcdsaSha2Nistp256Challenge {
+        hw_id,
+        sw_id: [0, 0, 0, 1],
```

Comment:
**question:** Is this equivalent to something like `SOFTWARE_IDENTITY.to_??_bytes()` for some (potentially inexistent) constant?

Reply:
This is mostly a placeholder for now, added a comment to that effect